src/main/java/org/cmps/tetrahedron/utils/ConvertTo3D.java

```diff
@@ -0,0 +1,51 @@
+package org.cmps.tetrahedron.utils;
+
+import org.cmps.tetrahedron.Main;
+import org.joml.*;
+import org.lwjgl.opengl.GL11C;
+
+import java.lang.Math;
+
+
+public class ConvertTo3D {
+
+    private Main main;
+    private final MouseControls mouseControls;
+
+    public ConvertTo3D(Main main, MouseControls mouseControls) {
+        this.main = main;
+        this.mouseControls = mouseControls;
+    }
+
+    public Vector3f convertScreenToWorld(double mouseX, double mouseY) {
+        Matrix4f viewProjMatrix = new Matrix4f(main.projMatrix).mul(main.viewMatrix);
+        float[] depth = new float[1];
+        GL11C.glReadPixels((int) mouseX, (int) (main.height - mouseY), 1, 1, GL11C.GL_DEPTH_COMPONENT, GL11C.GL_FLOAT, depth);
+
+        return viewProjMatrix.unproject((float) mouseX, (float) (main.height - mouseY), depth[0], new int[] {0, 0, main.width, main.height}, new Vector3f());
+    }
+
+    public void get3Dcoordinate() {
+        final double maxAcceptableDistanceSqr = 1e-3;
+        final double err = 1e-3;
+        double xMouse = mouseControls.getLastMouseX();
```

Comment:
Are you sure that for getting from mouse controls 2 coordinates, you need to have an instance of mouse controls in this class? Why not to pass them as method params?